core/lib/multivm/src/versions/vm_boojum_integration/tracers/circuits_tracer.rs

```diff
@@ -178,12 +205,17 @@ impl<S: WriteStorage, H: HistoryMode> VmTracer<S, H> for CircuitsTracer<S> {
             .precompile_cycles_history
             .inner();
         for (precompile, cycles) in &inner[last_precompile_inner_entry_checked..] {
-            let fraction = match precompile {
-                PrecompileAddress::Ecrecover => ECRECOVER_CYCLE_FRACTION,
-                PrecompileAddress::SHA256 => SHA256_CYCLE_FRACTION,
-                PrecompileAddress::Keccak256 => KECCAK256_CYCLE_FRACTION,
+            match precompile {
+                PrecompileAddress::Ecrecover => {
+                    self.statistics.ecrecover_cycles += *cycles as u32;
+                }
+                PrecompileAddress::SHA256 => {
+                    self.statistics.sha256_cycles += *cycles as u32;
+                }
+                PrecompileAddress::Keccak256 => {
+                    self.statistics.keccak256_cycles += *cycles as u32;
+                }
```

Comment:
just nit: 
I'd personally split it to 3 functions: 
1. Check precompiles
2. Check storage writes
3. Check opcodes